In [1]:
import numpy as np
import pandas as pd
import statsmodels.formula.api as smf
from scipy.stats import norm
import matplotlib.pyplot as plt

**1. Based on Standard Error**
$$
n=\left (\frac{\sigma}{\text{SE}(\bar{y})}  \right )^2
$$
  - $n$ is sample size estimate
  - $\sigma$ is standard deviation of population
  - $\text{SE}(\bar{y})$ is desired standard error

In [2]:
def sample_size_single_mean_se(sigma, standard_error):
    
    # Calculate sample size
    n = (sigma/standard_error)**2
    
    return n

**2. Based on Desired Power and Confidence Level**
- Sample size for estimating single mean based on power, confidence level and effectsize
$$
n = \frac{\sigma^2 (z_{1-{\frac{\alpha}{2}}}+z_{1-\beta})^2}{\delta^2}
$$
  - $n$ is sample size estimate
  - $\sigma$ is standard deviation of population
  - $\delta$ is desired difference to the true population
  - $z_{1-\frac{\alpha}{2}}$ represents z-value under desired significance level ($\alpha$)
  - $z_{1-\beta}$ represents z-value under the desired power 

In [3]:
def sample_size_single_mean(alpha, beta, effect_size, sigma):
    
    # Calculate z-value
    z_alpha = norm.ppf(1 - alpha/2)
    z_beta = norm.ppf(1 - beta)
    
    # Calculate sample size
    n = ( (sigma**2) * (z_alpha + z_beta)**2 ) / effect_size**2
    
    return n

- A real estate firm is evaluating a potential expansion into a high-value residential area.
- The firm's profitability depends on the property prices in the area they manage.
- In order to make the expansion financially feasible, the firm must ensure that the average property price in this area is more than `1,800,000`.
- From the Housing dataset, the standard deviation of property prices is approximately `1,870,000` (used as the population standard deviation estimate).

How much estimated the sample size if they want standard error to be no worse than 30,000?

In [4]:
# Given

standard_error = 30
sigma = 1870

In [5]:
# then calculate the number of sample size
n =  sample_size_single_mean_se(sigma, standard_error)

print(f"Number of sample size needed            : {n:.2f}")

Number of sample size needed            : 3885.44


If they wanted to conduct statistical inference to help the decision, how much estimated the sample size if they want the estimate has 80% power to be statistical significant under 95% confidence level to detect a difference of 30,000 from the true population mean?

Initiate what we know

In [6]:
# Given
effect_size = 30 # absolute desired difference with the true population mean
conf = 0.95      # confidence level
power = 0.8      # power
sigma = 1870   # standard deviation of population (assumed / approximated)

Calculate the alpha and beta from confidence level and power level

In [7]:
# calculate beta
beta = 1 - power

# calculate alpha
alpha = 1 - conf

Calculate estimated sample size using helper function, or you can manually calculate it by using calculator, it must expect approximately same value

In [8]:
# then calculate the number of sample size
n =  sample_size_single_mean(alpha, beta, effect_size, sigma)

print(f"Number of sample size needed            : {n:.2f}")

Number of sample size needed            : 30496.39


**Conclusion**

The estimated sample size to have a standard error of 30,000 in our inferences is:
- We need **3,885** observations

The estimated sample size to have 95% confidence and 80% power to detect a difference of 30,000 from the true population mean is:
- We need **30,496** $\approx$ **30,500** observations